workflow/controller/operator.go

```diff
@@ -275,14 +280,22 @@ func (woc *wfOperationCtx) persistUpdates() {
 		return
 	}
 	wfClient := woc.controller.wfclientset.ArgoprojV1alpha1().Workflows(woc.wf.ObjectMeta.Namespace)
-	_, err := wfClient.Update(woc.wf)
+	err := woc.checkAndCompress()
 	if err != nil {
-		woc.log.Warnf("Error updating workflow: %v", err)
+		woc.log.Warnf("Error compressing workflow: %v", err)
+	}
+	if woc.wf.Status.CompressedNodes != "" {
+		woc.clearNodeStatusMap()
+	}
+
+	_, err = wfClient.Update(woc.wf)
+	if err != nil {
+		woc.log.Warnf("Error updating workflow: %v %s", err, apierr.ReasonForError(err))
 		if argokubeerr.IsRequestEntityTooLargeErr(err) {
 			woc.persistWorkflowSizeLimitErr(wfClient, err)
 			return
 		}
-		if !apierr.IsConflict(err) {
+		if err != nil && !apierr.IsConflict(err) {
```

Comment:
Why was `err != nil` check added? We already know this to be true because we are in the `if err != nil` block